provider/pkg/provider/remote/commandOutputs.go

```diff
@@ -59,35 +59,22 @@ func (c *CommandOutputs) run(ctx p.Context, cmd string) (string, string, error)
 		session.Stdin = strings.NewReader(*c.Stdin)
 	}
 
-	stdoutr, stdoutw, err := os.Pipe()
-	if err != nil {
-		return "", "", err
-	}
-	stderrr, stderrw, err := os.Pipe()
-	if err != nil {
-		return "", "", err
-	}
-	session.Stdout = stdoutw
-	session.Stderr = stderrw
-
-	var stdoutbuf bytes.Buffer
-	var stderrbuf bytes.Buffer
-
-	stdouttee := io.TeeReader(stdoutr, &stdoutbuf)
-	stderrtee := io.TeeReader(stderrr, &stderrbuf)
+	var stdoutbuf, stderrbuf, stdouterrbuf bytes.Buffer
+	r, w := io.Pipe()
+	session.Stdout = io.MultiWriter(&stdoutbuf, &stdouterrbuf, w)
+	session.Stderr = io.MultiWriter(&stderrbuf, &stdouterrbuf, w)
 
-	stdoutch := make(chan struct{})
-	stderrch := make(chan struct{})
-	go util.CopyOutput(ctx, stdouttee, stdoutch, diag.Debug)
-	go util.CopyOutput(ctx, stderrtee, stderrch, diag.Info)
+	stdouterrch := make(chan struct{})
+	go util.CopyOutput(ctx, r, stdouterrch, diag.Info)
 
 	err = session.Run(cmd)
 
-	stdoutw.Close()
-	stderrw.Close()
+	w.Close()
+	<-stdouterrch
 
-	<-stdoutch
-	<-stderrch
+	if err != nil {
+		return "", "", fmt.Errorf("%w: running %q:\n%s", err, cmd, stdouterrbuf.String())
+	}
 
-	return stdoutbuf.String(), stderrbuf.String(), err
+	return strings.TrimSuffix(stdoutbuf.String(), "\n"), strings.TrimSuffix(stderrbuf.String(), "\n"), nil
```

Comment:
It looks to me that we're stripping newlines now.

Reply:
Yeah - this was to align behavior with local command which does the same. I believe this difference was accidental in the original implementation.